default/python/AI/ResourcesAI.py

```diff
@@ -159,6 +147,24 @@ def bake_future_focus(self, pid, focus, update=True, force=False):
             self.baked_planet_info[pid] = self.raw_planet_info.pop(pid)
         return success
 
+    def check_avoid_change_due_to_bureaucracy(self, pinfo, focus, focus_gain, influence_gain) -> bool:
+        if not self.bureaucracy or focus == pinfo.current_focus:
+            return False  # No bureaucracy or no change
+        last_turn = fo.currentTurn() - 1
+        if pinfo.planet.LastTurnColonized == last_turn or pinfo.planet.LastTurnConquered == last_turn:
+            return False  # no penalty for newly settled/conquered planets
+        threshold = 3.0
+        if (
```

Comment:
You could use `return any()`

Maybe you could try to extract each group to the variable? 
Naming would be a challenge here.

```
not_enought_influence = (focus == INFLUENCE and influence_gain < focus_gain + threshold)
...focus... = (pinfo.current_focus == INFLUENCE and focus_gain < influence_gain + threshold)
```

Reply:
This part was actually not working as intended, so I had to partially rewrite it.
The whole module is on my list of things that should be rewritten. E.g. it still assumes that changing e.g. from research to industry focus only decreases RP and increases PP, ignoring the effects of stability.